src/Components/Header/Header.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  addTodo: (value: string) => void,
+  isLoading: boolean,
+  setToggleAllTodos: () => void
+};
+
+export const Header: React.FC<Props> = ({
+  todos, addTodo, isLoading, setToggleAllTodos,
+}) => {
+  const [value, setValue] = useState('');
+
+  const formSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    addTodo(value);
+    setValue('');
+  };
+
+  const toggleAllTodos = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setToggleAllTodos();
+  };
+
+  // eslint-disable-next-line no-console
+  console.log(isLoading);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className={classNames(
+          'todoapp__toggle-all',
+          { active: todos.some(todo => !todo.completed) },
```

Comment:
don't call the function from the template, because it's it will be called every time when the state changed
so, make sense to move it to the variable with useMemo